homeassistant/components/climate/__init__.py

```diff
@@ -124,6 +126,7 @@
             vol.Exclusive(ATTR_TEMPERATURE, "temperature"): vol.Coerce(float),
             vol.Inclusive(ATTR_TARGET_TEMP_HIGH, "temperature"): vol.Coerce(float),
             vol.Inclusive(ATTR_TARGET_TEMP_LOW, "temperature"): vol.Coerce(float),
+            vol.Optional(ATTR_TEMPERATURE_UNIT): vol.Coerce(UnitOfTemperature),
```

Comment:
I think this change needs an architecture discussion.

Reply:
I already thought that 🙈  will open a discussion 👍

Reply:
- https://github.com/home-assistant/architecture/discussions/1177